Correct typos

vivekmore opened the pull request:
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

<!--
Please also reference the issue number in a commit message to [automatically close the related Github issue](https://help.github.com/articles/closing-issues-via-commit-messages/)

Note: It is also possible to add `[skip ci]` to your commit message to skip Travis tests
-->

vivekmore commented:
Makes sense.
so what do you suggest? 
submit PR on separate 7.x branch?

murdos commented:
> 
> 
> Makes sense.
> so what do you suggest?
> submit PR on separate 7.x branch?

There's no 7.x branch, we only have maintenance branches.
I've added the 7.x label, I guess the PR will stay open until we start to work on 7.x

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/jhipster/generator-jhipster?pullRequest=10788) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/jhipster/generator-jhipster?pullRequest=10788) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/jhipster/generator-jhipster?pullRequest=10788) it.</sub>

jdubois commented:
Let's first do the "easy" typos, that are not used by the sub-generators. For the others, it's going to be difficult.

vivekmore commented:
> Let's first do the "easy" typos, that are not used by the sub-generators. For the others, it's going to be difficult.

How about a gradual upgrade by keeping both variants like follows:

```
methodWithoutTypo() {
   ...
   // actual code
   ...
}

methodWITHtypo() {
   this.log.warn('Please use methodWithoutTypo() as this method will be deprecated in next major release');
   methodWithoutTypo();
}
```

MathieuAA commented:
@vivekmore that's a good idea! Can you add a comment to the "functions with typos" that says
```
@deprecated
To be removed for JHipster v7
```

That way, we could clean them up for the next major

murdos commented:
> Let's first do the "easy" typos, that are not used by the sub-generators. For the others, it's going to be difficult.

These will still be breaking changes, in particular for blueprints.

For example, if a blueprint is overriding a particular step (`setupClientconsts` in the example below), it won't be overridden anymore if we rename it in main generator to `setupClientConstants`:  
```
get writing() {
        const phaseFromJHipster = super._writing();
        const customPhaseSteps = {
            setupClientconsts() {
                // Do things that change generator-jhipster behavior...
            },
        };
        return { ...phaseFromJHipster, ...customPhaseSteps };
    }
```

Development on V7 will start soon, let's wait a bit more and don't break V6 for blueprints (and there're a more and more)

vivekmore commented:
@murdos - You're right
However, to get this small PR merged I've removed the breaking changes related to setup*() functions.

I'll make another PR as soon as v7 development starts that:
- removes deprecated methods from this PR
- corrects the setup*() methods (that I removed from my most recent update to this PR)

does that seem reasonable?

murdos commented on generators/app/index.js:
```diff
@@ -448,7 +448,7 @@ module.exports = class extends BaseBlueprintGenerator {
             },
 
-            askFori18n: prompts.askFori18n,
+            askForI18n: prompts.askForI18n,
```
```suggestion
            askFori18n: prompts.askForI18n,
```

murdos commented on generators/client/index.js:
```diff
@@ -189,7 +189,7 @@ module.exports = class extends BaseBlueprintGenerator {
             askForModuleName: prompts.askForModuleName,
             askForClient: prompts.askForClient,
-            askFori18n: prompts.askFori18n,
+            askForI18n: prompts.askForI18n,
```
```suggestion
            askFori18n: prompts.askForI18n,
```

murdos commented on generators/server/index.js:
```diff
@@ -348,7 +348,7 @@ module.exports = class extends BaseBlueprintGenerator {
             askForServerSideOpts: prompts.askForServerSideOpts,
             askForOptionalItems: prompts.askForOptionalItems,
-            askFori18n: prompts.askFori18n,
+            askForI18n: prompts.askForI18n,
```
```suggestion
            askFori18n: prompts.askForI18n,
```

vivekmore commented:
> I highlighted the 3 remaining breaking changes that should be reverted. Then this PR could be merged ;)

Good catch @murdos ! Sorry for missing these, and thank you for being patient.
I realize I have a few more stuff to learn about things exported from Generators.
